src/keyring.ts

```diff
@@ -135,9 +136,9 @@ export type Keyring<State extends Json> = {
    */
   signTransaction?(
     address: Hex,
-    transaction: Transaction,
+    transaction: TypedTransaction,
```

Comment:
I'm wondering whether this type is wide enough. In the [Trezor keyring TypeScript migration](https://github.com/MetaMask/eth-trezor-keyring/pull/161/files#diff-676874abf521692d7bfaefb494cbb22090466e343b9a0728cb3e3826fbc2bc5aR303), we had to use a type union of the old and new types of transactions because our code was still passing in both types.

The main difference between the two (as far as the keying was concerned at least) was the method of getting the `chainId` from the transaction. (`tx.getChainId()` v.s. `tx.common.chainId()`)

Reply:
I see, so we would need this import too,

```js
import type OldEthJsTransaction from 'ethereumjs-tx';
```

Maybe is better to add it now and avoid a problem later on

Reply:
Worth double-checking to see which type we need. I suspect the Trezor keyring might be using both just as a way to make migrating easier. We might not be using both types actively at one time.

Reply:
Mobile is using `TypedTransaction` from `@ethereumjs/tx` and extension is not using any type so far. Should be safe to only use `TypedTransaction` then.

Reply:
I was more concerned about the actual transaction object passed in, rather than the type used. But I double-checked and confirmed that both extension and mobile transaction controllers are passing in `@ethereumjs/tx`-generated transactions, so we should be good to just use `TypedTransaction`.

Reply:
Thanks for double checking @Gudahtt!